ocaml/xe-cli/newcli.ml

```diff
@@ -435,6 +435,38 @@ let assert_filename_permitted ?(permit_cwd = false) permitted_filenames filename
   | _ ->
       ()
 
+let do_http_get ofd url exit_code f =
+  try
+    let rec doit url =
+      let server, path = parse_url url in
+      debug "Opening connection to server '%s' path '%s'\n%!" server path ;
+      with_open_tcp server @@ fun (ic, oc) ->
+      Printf.fprintf oc "GET %s HTTP/1.0\r\n\r\n" path ;
+      flush oc ;
+      (* Get the result header immediately *)
+      let resultline = input_line ic in
+      debug "Got %s\n%!" resultline ;
+      match http_response_code resultline with
+      | 200 ->
+          f ic ; marshal ofd (Response OK)
+      | 302 ->
+          let headers = read_rest_of_headers ic in
+          let newloc = List.assoc "location" headers in
+          (* see above about Unixfd.with_connection *)
+          close_in_noerr ic ; close_out_noerr oc ; doit newloc
+      | _ ->
+          failwith "Unhandled response code"
+    in
+    doit url
+  with
+  | ClientSideError msg ->
+      marshal ofd (Response Failed) ;
+      Printf.fprintf stderr "Operation failed. Error: %s\n" msg ;
```

Comment:
Does the `msg` contains detailed info like `url`? If not then it might be helpful to log more details like:
`Printf.fprintf stderr "HTTP GET %s failed. Error: %s\n" url msg ;`

Reply:
The `url` has been logged in line#442 and even earlier in a more general timing before matching `cmd` in the main loop.